--- ltmain.sh	2009-07-09 18:26:08.481171963 +0200
+++ ltmain-fixed.sh	2009-07-09 18:26:00.203672045 +0200
@@ -5347,19 +5347,19 @@
 	    # It is a libtool convenience library, so add in its objects.
 	    convenience="$convenience $ladir/$objdir/$old_library"
 	    old_convenience="$old_convenience $ladir/$objdir/$old_library"
+			tmp_libs=
+			for deplib in $dependency_libs; do
+				deplibs="$deplib $deplibs"
+				if $opt_duplicate_deps ; then
+					case "$tmp_libs " in
+					*" $deplib "*) specialdeplibs="$specialdeplibs $deplib" ;;
+					esac
+				fi
+				tmp_libs="$tmp_libs $deplib"
+			done
 	  elif test "$linkmode" != prog && test "$linkmode" != lib; then
 	    func_fatal_error "\`$lib' is not a convenience library"
 	  fi
-	  tmp_libs=
-	  for deplib in $dependency_libs; do
-	    deplibs="$deplib $deplibs"
-	    if $opt_duplicate_deps ; then
-	      case "$tmp_libs " in
-	      *" $deplib "*) specialdeplibs="$specialdeplibs $deplib" ;;
-	      esac
-	    fi
-	    tmp_libs="$tmp_libs $deplib"
-	  done
 	  continue
 	fi # $pass = conv
 
